### DISCUSSION TITLE
Re-use the uploaded media with the id.

### DISCUSSION THREAD

**Discussion by muhitrhn** [plhery/node-twitter-api-v2#405] [Q&A]
Is it possible to re-use the media id forever? Currently I save the media id got trough `v1.uploadMedia()` but sometimes the API returns an error saying I have provided invalid media id and then I have to change/re-upload the media.

**Comment by YasharF** [comment]
The media_id returned from the Twitter APIs has an expiration.  The following is the related documentation:

https://developer.twitter.com/en/docs/twitter-api/v1/media/upload-media/api-reference/post-media-upload#:~:text=The%20response%20provides%20a%20media%20identifier%20in%20the,and%20media_key%20are%20only%20valid%20for%20expires_after_secs%20seconds.
